remoteworkitem/trackeritem_repository.go

```diff
@@ -5,6 +5,14 @@ import "github.com/jinzhu/gorm"
 // upload imports the items into database
 func upload(db *gorm.DB, tqID int, item map[string]string) error {
 	ti := TrackerItem{Item: item["content"], RemoteItemID: item["id"], BatchID: item["batch_id"], TrackerQueryID: uint64(tqID)}
-	err := db.Create(&ti).Error
+	db.Where("remote_item_id = ?", item["id"]).Find(&ti)
+	var err error
+	if ti.ID == 0 {
+		ti1 := TrackerItem{Item: item["content"], RemoteItemID: item["id"], BatchID: item["batch_id"], TrackerQueryID: uint64(tqID)}
+		err = db.Create(&ti1).Error
+	} else {
+		ti2 := TrackerItem{ID: ti.ID, Item: item["content"], RemoteItemID: item["id"], BatchID: item["batch_id"], TrackerQueryID: uint64(tqID)}
```

Comment:
```go
func upload(db *gorm.DB, tqID int, item map[string]string) error {
	remoteID := item["id"]
	batchID := item["batch_id"]
	content := item["content"]

	var ti TrackerItem
	if db.Where("remote_item_id = ? and tracker_query_id = ?", remoteID, tqID).Find(&ti).RecordNotFound() {
		ti = TrackerItem{
			Item:           content,
			RemoteItemID:   remoteID,
			BatchID:        batchID,
			TrackerQueryID: uint64(tqID)}
		return db.Create(&ti).Error
	}
	ti.Item = content
	ti.BatchID = batchID
	return db.Save(&ti).Error
}
```

Try to fetch the item based on remote_id AND tracker_query_id (to ensure it originally was created by the same query, in case 2 remote systems have the same id), if not found create a new one, else update content and batch id.